quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs

```diff
@@ -0,0 +1,335 @@
+};
+use quickwit_proto::tonic;
+use quickwit_proto::tonic::Request;
+use tokio_stream::StreamExt;
+use warp::{Filter, Rejection};
+
+use crate::jaeger_api::model::{
+    JaegerError, JaegerResponseBody, JaegerSearchBody, TracesSearchQueryParams,
+};
+use crate::json_api_response::JsonApiResponse;
+use crate::{require, with_arg, BodyFormat};
+
+/// Setup Jaeger API handlers
+///
+/// This is where all Jaeger handlers
+/// should be registered.
+/// Request are executed on the `otel traces v0_6` index.
+pub(crate) fn jaeger_api_handlers(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_services_handler(jaeger_service_opt.clone())
+        .or(jaeger_service_operations_handler(
+            jaeger_service_opt.clone(),
+        ))
+        .or(jaeger_traces_search_handler(jaeger_service_opt.clone()))
+        .or(jaeger_traces_handler(jaeger_service_opt.clone()))
+    // Register newly created handlers here.
+}
+
+/// GET otel-traces-v0_6/services
+pub fn jaeger_services_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_services_filter()
+        .and(require(jaeger_service_opt))
+        .then(jaeger_services)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+/// GET otel-traces-v0_6/services/{service}/operations
+pub fn jaeger_service_operations_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_service_operations_filter(jaeger_service_opt)
+        .then(jaeger_service_operations)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+/// GET otel-traces-v0_6/traces?service={service}&start={start_in_ns}&end={end_in_ns}&lookback=custom
+pub fn jaeger_traces_search_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_traces_search_filter(jaeger_service_opt)
+        .then(jaeger_traces_search)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+/// GET otel-traces-v0_6/traces/{trace-id-base-64}
+pub fn jaeger_traces_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_traces_filter(jaeger_service_opt)
+        .then(jaeger_get_trace)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+#[utoipa::path(
+    get,
+    tag = "Jaeger Services",
+    path = "/services",
+    responses(
+        (status = 200, description = "Successfully fetched services information.", body = TODO )
+    )
+)]
+pub(crate) fn jaeger_services_filter() -> impl Filter<Extract = (), Error = Rejection> + Clone {
+    warp::path!("otel-traces-v0_6" / "services").and(warp::get())
```

Comment:
2 issues:
- "services" should be prefix by "api" 
- we need to add a jaeger prefix to isolate those endpoints better.
 
So URL should be  "otel-traces-v0_6/jaeger/api/services"

I don't if it's possible but we could probably add a global prefix "otel-traces-v0_6/jaeger" in the parent handler

Reply:
```rust
    let jaeger_api_root_url = warp::path!("otel-traces-v0_6" / "jaeger" / ..);
    jaeger_api_root_url.and(
        jaeger_services_handler(jaeger_service_opt.clone())
        .or(jaeger_service_operations_handler(
            jaeger_service_opt.clone(),
        ))
        .or(jaeger_traces_search_handler(jaeger_service_opt.clone()))
        .or(jaeger_traces_handler(jaeger_service_opt.clone()))
    )

```

Reply:
we should even go for `warp::path!(OTEL_TRACES_INDEX_ID / "jaeger" / "api" / ..);`

Reply:
naive usage of the `const` gave me the following error
```
error[E0747]: constant provided when a type was expected
   --> quickwit-serve/src/jaeger_api/rest_handler.rs:104:17
    |
104 |     warp::path!(OTEL_TRACES_INDEX_ID / "jaeger" / "api" / "services").and(warp::get())
    |                 ^^^^^^^^^^^^^^^^^^^^

error[E0271]: type mismatch resolving `<And<And<And<And<And<And<impl Filter + FilterBase<Extract = (), Error = Infallible> + Copy, impl Filter + FilterBase<Extract = (...,), Error = ...> + Copy>, ...>, ...>, ...>, ...>, ...> as FilterBase>::Extract == ()`
   --> quickwit-serve/src/jaeger_api/rest_handler.rs:103:43
    |
103 | pub(crate) fn jaeger_services_filter() -> impl Filter<Extract = (), Error = Rejection> + Clone {
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `(_,)`, found `()`
    |
    = note:  expected tuple `(_,)`
            found unit type `()`

Some errors have detailed explanations: E0271, E0747.
For more information about an error, try `rustc --explain E0271`.
error: could not compile `quickwit-serve` (lib) due to 2 previous errors
```
I tried a couple of things (specifying static lifetime, wrapping in the `Box`) and nothing worked.

Reply:
agreed to proceed without `OTEL_TRACES_INDEX_ID`